packages/core/test/feedback/FeedbackForm.test.tsx

```diff
@@ -0,0 +1,111 @@
+    formTitle: 'Feedback Form',
+    namePlaceholder: 'Name',
+    emailPlaceholder: 'Email',
+    descriptionPlaceholder: 'Description',
+    submitButton: 'Submit',
+    cancelButton: 'Cancel',
+    errorTitle: 'Error',
+    formError: 'Please fill out all required fields.',
+    emailError: 'The email address is not valid.',
+  },
+};
+
+describe('FeedbackForm', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('renders correctly', () => {
+    const { getByPlaceholderText, getByText } = render(<FeedbackForm {...defaultProps} />);
+
+    expect(getByText(defaultProps.text.formTitle)).toBeTruthy();
+    expect(getByPlaceholderText(defaultProps.text.namePlaceholder)).toBeTruthy();
+    expect(getByPlaceholderText(defaultProps.text.emailPlaceholder)).toBeTruthy();
+    expect(getByPlaceholderText(defaultProps.text.descriptionPlaceholder)).toBeTruthy();
+    expect(getByText(defaultProps.text.submitButton)).toBeTruthy();
+    expect(getByText(defaultProps.text.cancelButton)).toBeTruthy();
+  });
+
+  it('shows an error message if required fields are empty', async () => {
+    const { getByText } = render(<FeedbackForm {...defaultProps} />);
+
+    fireEvent.press(getByText(defaultProps.text.submitButton));
+
+    await waitFor(() => {
+      expect(Alert.alert).toHaveBeenCalledWith(defaultProps.text.errorTitle, defaultProps.text.formError);
+    });
+  });
+
+  it('shows an error message if the email is not valid', async () => {
+    const { getByPlaceholderText, getByText } = render(<FeedbackForm {...defaultProps} />);
+
+    fireEvent.changeText(getByPlaceholderText(defaultProps.text.namePlaceholder), 'John Doe');
+    fireEvent.changeText(getByPlaceholderText(defaultProps.text.emailPlaceholder), 'not-an-email');
+    fireEvent.changeText(getByPlaceholderText(defaultProps.text.descriptionPlaceholder), 'This is a feedback message.');
+
+    fireEvent.press(getByText(defaultProps.text.submitButton));
+
+    await waitFor(() => {
+      expect(Alert.alert).toHaveBeenCalledWith(defaultProps.text.errorTitle, defaultProps.text.emailError);
+    });
+  });
+
+  it('calls captureFeedback when the form is submitted successfully', async () => {
+    const { getByPlaceholderText, getByText } = render(<FeedbackForm {...defaultProps} />);
+
+    fireEvent.changeText(getByPlaceholderText(defaultProps.text.namePlaceholder), 'John Doe');
+    fireEvent.changeText(getByPlaceholderText(defaultProps.text.emailPlaceholder), 'john.doe@example.com');
+    fireEvent.changeText(getByPlaceholderText(defaultProps.text.descriptionPlaceholder), 'This is a feedback message.');
+
+    fireEvent.press(getByText(defaultProps.text.submitButton));
+
+    await waitFor(() => {
+      expect(captureFeedback).toHaveBeenCalledWith({
+        message: 'This is a feedback message.',
+        name: 'John Doe',
+        email: 'john.doe@example.com',
+      });
+    });
+  });
+
+  it('calls closeScreen when the form is submitted successfully', async () => {
```

Comment:
Q: What happens if we send the feedback without internet?

Reply:
Good question @lucas-zimerman 👍 
I don't see an easy solution to this (or maybe there is? 🤔) since the [captureFeedback](https://github.com/getsentry/sentry-javascript/blob/67b8d3e55a3d2d53a6ff4e57251e73606c666a87/packages/core/src/feedback.ts#L8) does not really handle this. I was thinking of using an implementation [similar to sendFeedback of Sentry JS](https://github.com/getsentry/sentry-javascript/blob/develop/packages/feedback/src/core/sendFeedback.ts).
I opened an issue https://github.com/getsentry/sentry-react-native/issues/4359 to handle this separately since this PR has grown too big.

Reply:
Opened a PR for this https://github.com/getsentry/sentry-react-native/pull/4364